### PR TITLE
[PLAT-1085] Make backwards compatible changes using 2to3

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

In order to defray some of the big changes that will arise by switching over to Python3 we've decided to switch backwards compatible changes first so we don't have to go through the process all at once.

## Changes

- convert map function
- convert dicts methods
- convert iteritems/itertools methods
- convert filter functions
- convert reduce function
- convert some basestring types to str
- convert next function

## QA Notes

Not sure how you would test something so big! Passes unittests etc. 

## Documentation

Not a user facing change.

## Side Effects

We might want to break this up into smaller piece if it's a lot to test. The changes are mostly automatic and probably not breaking, but likely some areas aren't touched by unittests

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-1085

### CODE REVIEW

**Comment by sloria** [comment]
Aha! It looks like the dict fixer was run twice: first in https://github.com/CenterForOpenScience/osf.io/pull/8726/commits/d2dc159db46e769ecf071db0c8a9cd4991c7b12e then again in https://github.com/CenterForOpenScience/osf.io/pull/8726/commits/da6f0a2b3217f2e1b173fd5dbe49b6b458ef90eb , resulting in the `list()` wrapping.

@Johnetordoff Can you run all the fixers in one go?

**Comment by Johnetordoff** [comment]
The problem with running them all at once is that I've cherry-picked out some of the breaking conversions, but it shouldn't be hard to do again now that I know what they are.

**Comment by sloria** [comment]
@Johnetordoff It makes sense that `d.keys()` -> `list(d.keys())`. The problem was that this PR was doing `d.iterkeys()`-> `d.keys()` -> `list(d.keys())`, where `d.keys()` is the desired output. This was happening because 2to3 was run more than once.

File: addons/base/models.py [hunk 1]
```diff
@@ -294,11 +294,11 @@ def merge(self, user_settings):
         if user_settings.__class__ is not self.__class__:
             raise TypeError('Cannot merge different addons')
 
-        for node_id, data in user_settings.oauth_grants.iteritems():
+        for node_id, data in user_settings.oauth_grants.items():
             if node_id not in self.oauth_grants:
                 self.oauth_grants[node_id] = data
             else:
-                node_grants = user_settings.oauth_grants[node_id].iteritems()
+                node_grants = iter(user_settings.oauth_grants[node_id].items())
```
Comment (sloria) [inline addons/base/models.py]: Drive-by: The `iter` here is unnecessary.

File: addons/base/tests/views.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def test_oauth_start(self):
         redirect_params = urlparse.parse_qs(redirect_url.query)
         provider_url = urlparse.urlparse(self.Provider().auth_url)
         provider_params = urlparse.parse_qs(provider_url.query)
-        for param, value in redirect_params.items():
+        for param, value in list(redirect_params.items()):
```
Comment (sloria) [inline addons/base/tests/views.py]: As expected, `d.items()` -> `list(d.items()`, but it's unnecessary in this context.

@Johnetordoff Can you see if you can fix the places where it's unnecessary to coerce to a list?
